lua/luasnip/loaders/from_vscode.lua

```diff
@@ -187,12 +187,42 @@ function M.load(opts)
 		opts.paths = get_snippets_rtp()
 	end
 
+	opts.paths = vim.tbl_map(expand_path, opts.paths) -- Expand before deduping, fake paths will become nil
+	opts.paths = vim.tbl_keys(list_to_set(opts.paths)) -- Remove doppelgänger paths and ditch nil ones
+
 	for _, path in ipairs(opts.paths) do
-		local full_path = expand_path(path)
-		if full_path then
-			load_snippet_folder(full_path, opts)
+		load_snippet_folder(path, opts)
+	end
+end
+
+local lazy_load_paths = {}
+local lazy_loaded_ft = {}
+
+function _G._luasnip_vscode_lazy_load()
+	for _, ft in ipairs({ vim.bo.filetype, "all" }) do
+		if not lazy_loaded_ft[ft] then
+			lazy_loaded_ft[ft] = true
+			M.load({ paths = lazy_load_paths, include = { ft } })
 		end
 	end
 end
 
+function M.lazy_load(opts)
+	opts = opts or {}
+
+	-- We have to do this here too, because we have to store them in lozy_load_paths
+	if type(opts.paths) ~= "table" and opts.paths ~= nil then
+		opts.paths = vim.split(opts.paths, ",")
+	else
+		opts.paths = get_snippets_rtp()
+	end
+	vim.list_extend(lazy_load_paths, opts.paths)
+	vim.cmd([[
+		augroup _luasnip_vscode_lazy_load
+		autocmd!
+		au BufEnter * lua _G._luasnip_vscode_lazy_load()
```

Comment:
tbh I'd rather not have `lazy_load` in the global namespace, the user probably won't interact with it directly. Maybe put the function in `M` and call here using `require`?

Reply:
Ok, I put it on _G as it's not exposed API, but if you prefer not to clutter the global context,  I will do it.

Reply:
ty!